docs/guides/05-building/cross-platform.md

```diff
@@ -4,4 +4,76 @@ sidebar_position: 2
 
 # Cross-Platform Compilation
 
-How to use GH Action for Building: a glance at Tauri Action.
+## How to use GH Action for Building: a glance at Tauri Action
+
+### Introduction
+
+For cross-platform compilation Tauri provides Tauri Action, a GitHub Action that builds your Web application as a Tauri native binary for macOS, Linux, and Windows, it can also release the application on GitHub releases. To set up Tauri Action you must first set up a GitHub repository, you can use this action on a repo that doesn't have Tauri configured since we automatically initialize Tauri before building, and configure it to use your Web artifacts.
+
+Tauri Action has three main usages: test the build pipeline of your Tauri app, upload Tauri artifacts to an existing release, and create a new release with the Tauri artifacts.
+
+### How to set up
+
+Go to [Tauri Action GitHub page][1]. Then head to GitHub Action on your GitHub project and choose "New workflow", then choose "Set up a workflow yourself". Replace the file with the test example. You can do the same steps for the [production build workflow example][2].
+
+### Notes
+
+You may modify the workflow name, change the triggers, add more steps such as npm run lint or npm run test. The important part is that you keep the:
+
+```yaml
+- uses: tauri-apps/tauri-action@v0
+```
+
+line at the final of the workflow, since this runs the build script and releases the artifacts.
+
+Also note this line next to the end of the workflow:
+
+```yaml
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+```
+
+This GitHub Token is automatically issued by GitHub for each workflow run without further configuration, which means there is no risk of secret leakage. This token however only has read permissions by default and you may get a "Resource not accessible by integration" error when running the workflow, if this happens to fix it you may add read permissions to this token. To do this head over to your Project Settings, and then select Actions, scroll down until "Workflow permissions" and check "Read and write permissions".
+
+### How to trigger
+
+By default, the testing workflow is triggered on each pull request, but you could also trigger it for every push to the repository on all branches or select branches by changing the start like this:
```

Comment:
I think this guide should focus on the release aspect of the action, not the testing one

Reply:
Should this be moved to another guide, or completelly deferred to tauri-action documentation? I quite like using tauri-action for testing as it requires minimal configuration if you are already going to setup a release workflow so I though it would be nice to have it here. Maybe a new guide could be added on the testing section which explains this.